src/stream/src/executor/aggregation/mod.rs

```diff
@@ -410,6 +410,7 @@ pub fn generate_state_table<S: StateStore>(
     let table_desc = generate_column_descs(agg_call, group_keys, pk_indices, agg_schema, input_ref);
     // Always leave 1 space for agg call value.
     let relational_pk_len = table_desc.len() - 1;
+    let dist_keys: Vec<usize> = (0..group_keys.len()).collect();
```

Comment:
IIUC here `dist_keys` stands for the dist key indices in pk. But in fact we should pass **dist key indices in value** to state table? cc. @BugenZhao

Reply:
Hmmm, I remembered that when compute vnode we are using indices on pk?

Reply:
> But in fact we should pass dist key indices in value to state table?

Yes. However, it seems they are the same in this case?

Reply:
> Hmmm, I remembered that when compute vnode we are using indices on pk?

We switch to indices on all columns to reduce the complexity of the interfaces. The "indices based on pk" will be inferred in the cell-based table.

https://github.com/singularity-data/risingwave/blob/4cf6a163d4767f34e761ee6172ef93f039161cb9/src/storage/src/table/cell_based_table.rs#L84-L92